src/diart/models.py

```diff
@@ -143,33 +162,25 @@ def from_pyannote(
         assert _has_pyannote, "No pyannote.audio installation found"
         return PyannoteEmbeddingModel(model, use_hf_token)
 
-    @abstractmethod
-    def forward(
-        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
-    ) -> torch.Tensor:
-        """
-        Forward pass of an embedding model with optional weights.
-
-        Parameters
-        ----------
-        waveform: torch.Tensor, shape (batch, channels, samples)
-        weights: Optional[torch.Tensor], shape (batch, frames)
-            Temporal weights for each sample in the batch. Defaults to no weights.
-
-        Returns
-        -------
-        speaker_embeddings: torch.Tensor, shape (batch, embedding_dim)
-        """
-        pass
-
 
 class PyannoteEmbeddingModel(EmbeddingModel):
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__(PyannoteLoader(model_info, hf_token))
 
-    def forward(
-        self,
-        waveform: torch.Tensor,
-        weights: Optional[torch.Tensor] = None,
+    def __call__(
+        self, waveform: torch.Tensor, weights: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        return self.model(waveform, weights=weights)
+        if not isinstance(self.model, WeSpeakerPretrainedSpeakerEmbedding):
```

Comment:
This check isn't needed, we can normalize weights for any kind of embedding model. The original performance won't be affected. Please remove

Reply:
I added the normalization always now. Sadly, we still have separate branches for the `to("cpu")` part. Any idea how to get rid of them? I think the change has to happen in pyannote itself.